qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1060,38 +984,16 @@ def to_file(self, fp, samples=None):
             If supplied, only the specified samples will be written to the
             file
         """
-        conn_handler = SQLConnectionHandler()
-        metadata_map = self._transform_to_dict(conn_handler.execute_fetchall(
-            "SELECT * FROM qiita.{0} WHERE {1}=%s".format(self._table,
-                                                          self._id_column),
-            (self.id,)))
-        dyn_vals = self._transform_to_dict(conn_handler.execute_fetchall(
-            "SELECT * FROM qiita.{0}".format(self._table_name(self.id))))
-
-        for k in metadata_map:
-            for key, value in viewitems(self.translate_cols_dict):
-                id_ = metadata_map[k][key]
-                metadata_map[k][value] = self.str_cols_handlers[key][id_]
-                del metadata_map[k][key]
-            metadata_map[k].update(dyn_vals[k])
-            metadata_map[k].pop('study_id', None)
-
-        # Remove samples that are not in the samples list, if it was supplied
+        df = self.to_dataframe()
         if samples is not None:
-            for sid, d in metadata_map.items():
-                if sid not in samples:
-                    metadata_map.pop(sid)
-
-        # Write remaining samples to file
-        headers = sorted(list(metadata_map.values())[0].keys())
-        with open(fp, 'w') as f:
-            # First write the headers
-            f.write("sample_name\t%s\n" % '\t'.join(headers))
-            # Write the values for each sample id
-            for sid, d in sorted(metadata_map.items()):
-                values = [str(d[h]) for h in headers]
-                values.insert(0, sid)
-                f.write("%s\n" % '\t'.join(values))
+            df = df.loc[samples]
+
+        # Apply some sorting to the dataframe
```

Comment:
I guess I could parse the file using pandas and check at the parsed pandas dataframe level. Does this sound good to everybody? Note that the original intent of the test is to check the contents of the file, and I'm unsure if we are loosing this check in this way...